dao/elasticsearch/service.go

```diff
@@ -52,11 +79,25 @@ func (this *ControlPlaneDao) CloneService(request dao.ServiceCloneRequest, clone
 
 //
 func (this *ControlPlaneDao) UpdateService(svc service.Service, unused *int) error {
+	if gotLock, blocker := this.facade.DFSLock(datastore.Get()).LockWithTimeout("update service", userLockTimeout); !gotLock {
+		err := errors.New(fmt.Sprintf(userLockTimeoutMessage, blocker))
+		glog.Warningf("Cannot update service: %s", err)
+		return err
+	}
+	defer this.facade.DFSLock(datastore.Get()).Unlock()
+
 	return this.facade.UpdateService(datastore.Get(), svc)
 }
 
 //
 func (this *ControlPlaneDao) MigrateServices(request dao.ServiceMigrationRequest, unused *int) error {
+	if gotLock, blocker := this.facade.DFSLock(datastore.Get()).LockWithTimeout("migrate service", userLockTimeout); !gotLock {
+		err := errors.New(fmt.Sprintf(userLockTimeoutMessage, blocker))
```

Comment:
To create an error type in go, the struct you create has to implement Error() string